desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -115,11 +116,33 @@ def handle_on_link_shared(channel_id, message_ts, links):
       msg = "Document with {key}={value} does not exist".format(key='uuid' if id_type == 'uuid' else 'id', value=qid)
       raise PopupException(_(msg))
 
-    payload = _make_unfurl_payload(item['url'], id_type, doc, doc_type)
+    # Mock request for query execution and fetch result
+    user = rewrite_user(User.objects.get(username=doc.owner.username))
+    request = MockRequest(user=user)
+
+    payload, file_status = _make_unfurl_payload(request, item['url'], id_type, doc, doc_type)
     response = slack_client.chat_unfurl(channel=channel_id, ts=message_ts, unfurls=payload)
     if not response['ok']:
       raise PopupException(_("Cannot unfurl link"), detail=response["error"])
-
+    
+    # Generate and upload result xlsx file only if result available
+    if file_status:
+      notebook = json.loads(doc.data)
+      snippet = notebook['snippets'][0]
+      snippet['statement'] = notebook['snippets'][0]['statement_raw']
+      file_format = 'xlsx'
+      file_name = _get_snippet_name(notebook)
+
+      content_generator = get_api(request, snippet).download(notebook, snippet, file_format='xls')
+
+      response = slack_client.files_upload(
+        channels=channel_id, 
+        file=next(content_generator), 
+        thread_ts=message_ts,
```

Comment:
So it shows up in the thread only? (not bad for v1, just wondering why we could not send it as attachment)

Reply:
What does sending as an attachment means? Like directly sending in the channel as other message?? @romainr